articles/express/index.md

```diff
@@ -0,0 +1,360 @@
+author: rohan-reddy
+date: 2020-09-18T00:00:00-11:00
+topics: [Node.js]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/express/hero.jpg
+    alt: expressjs example text image
+---
+[Node.js](https://www.section.io/engineering-education/history-of-nodejs/) is a JavaScript run time environment which is used to create server-side applications and tools. 
+
+Node.js is fast, portable and written in JavaScript but it doesn't directly support common tasks such as handling requests, serving files and handling HTTP methods such as `GET` and `POST` but this is where Node.js's [rich ecosystem](https://www.section.io/engineering-education/most-useful-nodejs-packages/) comes to our aid.
+<!--more-->
+
+**Express.js (Express) is a light web framework which sits on top of Node.js and it adds functionality ([middleware](https://expressjs.com/en/guide/writing-middleware.html), [routing](https://en.wikipedia.org/wiki/Routing), etc.) and simplicity to Node.js.**
+
+When creating a Node.js web application, we write a single JavaScript application which listens to requests from the browser, based on the request, the function will send back some data or an HTML web page.
+
+![flow of a request](/engineering-education/express/node.png)
+
+[Image Credit](https://www.manning.com/books/express-in-action)
+
+A *request handler* is a JavaScript function which takes a request and sends an appropriate response.
+
+Node.js APIs can get complex and writing how to handle a single request can end up being over 50 lines of code. Express makes it easier to write Node.js web applications.
+
+![flow with express](/engineering-education/express/express.png)
+
+[Image Credit](https://www.manning.com/books/express-in-action)
+
+### Advantages of using Express with Node.js
+- Express lets you take away a lot of Node.js complexities while adding helpful functions to a Node.js HTTP server.
+
+- Instead of a large request handler function, Express allows us to handle requests by writing many small modular and maintainable functions.
+
+- Express is *not opinionated*, meaning Express does not enforce any "right way" of doing things. You can use any compatible middleware, and you can structure the app as you wish, making it flexible.
+
+- We can integrate with a [template rendering engine](https://www.digitalocean.com/community/tutorials/nodejs-express-template-engines)(also called as view rendering engine in some articles) of our choice like Jade, Pug, EJS, etc. 
+
+  A template engine enables you to use static template files and at runtime change the values of variables in those files.
+
+- You can set up ["middleware"](https://expressjs.com/en/guide/using-middleware.html) for request processing.
+
+### Basic Express App
+Install express via npm
+
+```bash
+$ npm install express --save
+```
+
+Let's create a basic Express example app.
+
+```JavaScript
+//app.js
```

Comment:
You need to tell readers you need to create the app.js file. Beginners may get stuck here.